burr/core/application.py

```diff
@@ -1127,38 +1181,152 @@ def with_hooks(self, *adapters: LifecycleAdapter) -> "ApplicationBuilder":
         return self
 
     def with_tracker(
-        self, project: str, tracker: Literal["local"] = "local", params: Dict[str, Any] = None
+        self,
+        tracker: Union[Literal["local"], object] = "local",
+        project: str = "default",
+        params: Dict[str, Any] = None,
     ):
         """Adds a "tracker" to the application. The tracker specifies
         a project name (used for disambiguating groups of tracers), and plugs into the
         Burr UI. Currently the only supported tracker is local, which takes in the params
         `storage_dir` and `app_id`, which have automatic defaults.
 
-        :param project: Project name
-        :param tracker: Tracker to use, currently only ``local`` is available
-        :param params: Parameters to pass to the tracker
+        :param tracker: Tracker to use. ``local`` creates one, else pass one in.
+        :param project: Project name -- used if the tracker is local.
+        :param params: Parameters to pass to the tracker if it's local.
         :return: The application builder for future chaining.
         """
         if params is None:
             params = {}
-        if tracker == "local":
+        # if it's a lifecycle adapter, just add it
+        if isinstance(
```

Comment:
See note above... Now that I'm looking at it, we should have a function that tells us whether something is a hook so we don't have to do this. This feels a bit ugly -- there's an implicit contract here that we don't make explicit. 

Also, why these specific ones? Have a comment/explanation.